### PR TITLE
Add missing fields to structs

### PR DESCRIPTION
**Pull request by kushmansingh**
There fields specified in the API docs that are missing from this client lib, added them after verifying they existed via curl

### CODE REVIEW

File: buildkite/pipelines.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ type CreatePipeline struct {
 	Steps      []Step `json:"steps" yaml:"steps"`
 
 	// Optional fields
+	Configuration                   string            `json:"configuration,omitempty" yaml:"configuration,omitempty"`
```
Comment (ticky) [inline buildkite/pipelines.go:23]: With YAML steps enabled, you supply either Steps or Configuration, so I believe these both need to be optional now:

```suggestion

	// Optional fields
	Steps                           []Step            `json:"steps,omitempty" yaml:"steps,omitempty"`
	Configuration                   string            `json:"configuration,omitempty" yaml:"configuration,omitempty"`
```

File: buildkite/pipelines.go [hunk 2]
```diff
@@ -168,22 +178,7 @@ func (ps *PipelinesService) Update(org string, p *Pipeline) (*Response, error) {
...
-	if p.Provider != nil {
-		cp.ProviderSettings = p.Provider.Settings
-	}
-
-	for i := range p.Steps {
-		cp.Steps[i] = *p.Steps[i]
-	}
-
-	req, err := ps.client.NewRequest("PATCH", u, cp)
+	req, err := ps.client.NewRequest("PATCH", u, p)
```
Comment (ticky) [inline buildkite/pipelines.go:181]: I think this works, I’m not sure what properties that comment was talking about. 👍🏼

**Review by ticky — CHANGES_REQUESTED** [review]
Looking great, I think there’s one change which would be helpful

**Comment by sj26** [comment]
Neat!

Will this break if we add new fields to these responses?

File: buildkite/pipelines.go [hunk 3]
```diff
@@ -17,9 +17,13 @@ type PipelinesService struct {
 type CreatePipeline struct {
 	Name       string `json:"name" yaml:"name"`
 	Repository string `json:"repository" yaml:"repository"`
-	Steps      []Step `json:"steps" yaml:"steps"`
+
+	// Either configuration needs to be specified as a yaml string or steps.
+	Configuration string `json:"configuration,omitempty" yaml:"configuration,omitempty"`
+	Steps         []Step `json:"steps" yaml:"steps"`
```
Comment (sj26) [inline buildkite/pipelines.go:23]: Should this be modified to `omitempty` as well?

Comment (jayco) [inline buildkite/pipelines.go:23]:
```suggestion
	Steps         []Step `json:"steps, omitempty" yaml:"steps"`
```

**Comment by JuanitoFatas** [comment]
@kushmansingh Thanks Kush so much for your contributions ❤️  I will cut a release soon.

**Pull request closed without merging**